docs/Makefile

```diff
@@ -3,7 +3,7 @@
 
 # You can set these variables from the command line, and also
 # from the environment for the first two.
-SPHINXOPTS    ?=
+SPHINXOPTS    ?= -W
```

Comment:
[nit] add `--keep-going` so the user gets the complete list of warnings rather than failing and stopping on the first warning.